akka-http/src/main/scala/akka/http/scaladsl/coding/Decoder.scala

```diff
@@ -13,13 +13,19 @@ import headers.HttpEncoding
 import akka.stream.scaladsl.{ Flow, Sink, Source }
 
 import scala.concurrent.Future
+import scala.util.control.NonFatal
 
 trait Decoder {
   def encoding: HttpEncoding
 
   def decodeMessage(message: HttpMessage): message.Self =
     if (message.headers exists Encoder.isContentEncodingHeader)
-      message.transformEntityDataBytes(decoderFlow).withHeaders(message.headers filterNot Encoder.isContentEncodingHeader)
+      message.transformEntityDataBytes(decoderFlow.recover {
+        case NonFatal(e) ⇒
```

Comment:
Not sure if that catch all exception handler is really warranted (but you didn't create it anyway, so it's fine for now but maybe create another ticket to investigate if we can be more precise, so that not all other upstream errors are converted here.